kinesis-video-gstreamer-plugin/src/gstkvssink.cpp

```diff
@@ -1274,7 +1290,27 @@ init_track_data(GstKvsSink *kvssink) {
                 kvs_sink_track_data->track_id = KVS_SINK_DEFAULT_AUDIO_TRACKID;
             }
 
-            audio_content_type = g_strdup(DEFAULT_CONTENT_TYPE_AUDIO);
+            GstCollectData *collect_data = (GstCollectData *) walk->data;
+
+            // extract media type from GstCaps to check whether it's h264 or h265
+            caps = gst_pad_get_allowed_caps(collect_data->pad);
+            media_type = gst_structure_get_name(gst_caps_get_structure(caps, 0));
+            if (strncmp(media_type, GSTREAMER_MEDIA_TYPE_AAC, MAX_GSTREAMER_MEDIA_TYPE_LEN) == 0) {
+                // default codec id is for aac audio.
+                audio_content_type = g_strdup(DEFAULT_CONTENT_TYPE_AUDIO_AAC);
+            } else if (strncmp(media_type, GSTREAMER_MEDIA_TYPE_ALAW, MAX_GSTREAMER_MEDIA_TYPE_LEN) == 0) {
+                g_free(kvssink->audio_codec_id);
+                kvssink->audio_codec_id = g_strdup(DEFAULT_AUDIO_CODEC_ID_PCM);
+                audio_content_type = g_strdup(DEFAULT_CONTENT_TYPE_AUDIO_ALAW);
+            } else if (strncmp(media_type, GSTREAMER_MEDIA_TYPE_MULAW, MAX_GSTREAMER_MEDIA_TYPE_LEN) == 0) {
+                g_free(kvssink->audio_codec_id);
+                kvssink->audio_codec_id = g_strdup(DEFAULT_AUDIO_CODEC_ID_PCM);
+                audio_content_type = g_strdup(DEFAULT_CONTENT_TYPE_AUDIO_MULAW);
+            } else {
+                // no-op, should result in a caps negotiation error before getting here.
+                LOG_AND_THROW("Code should not reach here");
```

Comment:
LOL: the famous last words. At any rate, change this to an "Internal error with blah blah"